internal/errors/fatal.go

```diff
@@ -7,25 +7,37 @@ import (
 
 // fatalError is an error that should be printed to the user, then the program
 // should exit with an error code.
-type fatalError string
+type fatalError struct {
+	msg      string
+	innerErr error
+}
 
 func (e fatalError) Error() string {
-	return string(e)
+	return e.msg
+}
+
+func (e fatalError) Unwrap() error {
+	return e.innerErr
 }
 
 // IsFatal returns true if err is a fatal message that should be printed to the
 // user. Then, the program should exit.
 func IsFatal(err error) bool {
-	var fatal fatalError
+	var fatal *fatalError
 	return errors.As(err, &fatal)
 }
 
 // Fatal returns an error that is marked fatal.
 func Fatal(s string) error {
-	return Wrap(fatalError(s), "Fatal")
+	return &fatalError{fmt.Sprintf("Fatal: %v", s), nil}
 }
 
 // Fatalf returns an error that is marked fatal.
 func Fatalf(s string, data ...interface{}) error {
-	return Wrap(fatalError(fmt.Sprintf(s, data...)), "Fatal")
+	return FatalWrap(fmt.Errorf(s, data...))
+}
+
+// FatalWrap wraps a lesser error as fatal.
+func FatalWrap(err error) error {
+	return &fatalError{fmt.Sprintf("Fatal: %v", err), err}
```

Comment:
`fmt.Errorf` does provide error wrapping using the `%w` format specifier used like `fmt.Errof("Fatal: %w", err)`

Reply:
Yes, it does - but the result wouldn't be a `fatalError` type, which parts of the code elsewhere check for - `fmt.Errorf` wraps using an internal stdlib type. So if you want wrapping with a specific type (`fatalError`), you need to kind of do that yourself.

(at least, that is my limited understanding of Go)